x-pack/plugins/security_solution/cypress/tasks/create_new_rule.ts

```diff
@@ -299,6 +313,15 @@ export const waitForTheRuleToBeExecuted = () => {
   });
 };
 
-export const selectEqlRuleType = () => {
-  cy.get(EQL_TYPE).click({ force: true });
+export const waitForAlertsToPopulate = async () => {
+  cy.waitUntil(() => {
+    refreshPage();
+    return cy
+      .get(NUMBER_OF_ALERTS)
+      .invoke('text')
+      .then((countText) => {
+        const alertCount = parseInt(countText, 10) || 0;
+        return alertCount > 0;
+      });
+  });
```

Comment:
Nice!

For the backend e2e, I started just like this, but then later I had to add a specific number to wait for a specific number of alerts as even with things like `refresh=wait_for` within the code there does exist some time before the Nth alert shows up.

I had some flakiness where 1 signal would show up on the backend but I needed to wait for like 2 or 3. No changes requested, but if we see flaky we will probably have to add that integer here.